### ISSUE TITLE
Provide fallback storage when using tus in Web Workers

### ISSUE THREAD

**Issue by dumitrascuSorin** [tus/tus-js-client#664]
**Is your feature request related to a problem? Please describe.**
Following PR #660, uploading files from a Web Worker has been a breeze. 
However, resuming an interrupted upload is no longer possible because of the hard dependency on localStorage, which is not supported in Web Workers.

**Describe the solution you'd like**
Would be nice to use a storage mechanism supported by Web Workers, such as the IndexedDB API.

**Can you provide help with implementing this feature?**
Yes, if the tus team is open to this idea.

Looking forward to your reply.

**Comment by Acconut** [comment]
Good idea, we would be happy to accept a PR adding an IndexedDB-based URL storage to tus-js-client.

**Comment by geekypradip** [comment]
> Good idea, we would be happy to accept a PR adding an IndexedDB-based URL storage to tus-js-client.

I've implemented a new feature for my personal project. Previously, I encountered issues when uploading more than 30,000 files at once. The problem stemmed from local storage limitations, causing it to fill up quickly and ultimately crash the browser. To address this, I integrated indexDB-based storage into the system. Now, the application checks whether the client's browser supports indexDB. If it does, the system utilizes indexDB; otherwise, it falls back to local storage or noopStorage.

But not using web-workers

**Comment by Acconut** [comment]
@geekypradip Thank you for the PR. Anybody who is interested can have a look at #677

**Comment by kesavkolla** [comment]
If we write the file to OPFS then tus can guarantee even browser restart will continue upload.

In general OPFS read/write are faster than Indexdb. Indexdb again depends on the async loop and also has to do serialize and deserialize the data. 

OPFS is native file I/O so read write doesn't involve any serialization.

**Comment by Acconut** [comment]
Thanks for the insights!

> If we write the file to OPFS then tus can guarantee even browser restart will continue upload.

Do you know if that's not the case for IndexedDB? Does data in IndexedDB get erased when the browser closes?

**Comment by kesavkolla** [comment]
Indexdb will keep data after browser restarts. But yeah performance of file I/O is better than Indexdb.
